nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITDirectory.java

```diff
@@ -100,4 +102,45 @@ public void itCanCreateUpdateFetchDeleteDirectory() {
         directory.deleteEntry(result.getId());
     }
 
+    @Test
+    public void itCanSetOrderingPropertyInteger() {
```

Comment:
Replying to your question here, as I can't reply to the discussion:
>When trying to implement this change, I am getting a cast error. Not sure if it is a problem with the code or I've written the test incorrectly?

Don't hesitate to put more information, like the stack trace you get for instance.

Here, I suppose this is something related to the code, I suppose that because if there is a problem in the test you would get an assertion failure and not a class cast, which is an exception that shouldn't happen.

Reply:
After implementing the changes, I'm still getting the errors :
`java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')

	at org.nuxeo.client.objects.directory.DirectoryEntry.putOrderingProperty(DirectoryEntry.java:156)
	at org.nuxeo.client.ITDirectory.itCanSetOrderingPropertyLong(ITDirectory.java:145)
	
	java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')

	at org.nuxeo.client.objects.directory.DirectoryEntry.putOrderingProperty(DirectoryEntry.java:156)
	at org.nuxeo.client.ITDirectory.itCanSetOrderingPropertyInteger(ITDirectory.java:119)`
	
	When debugging it seems that `entryWithInteger` has Long 0 as the property but once `resultWithInteger` is created, it changes to an Integer 0.

Reply:
Ok, this is probably due to deserialization with Jackson which takes an Integer type as the number is not in the Long range.

This means that the `putOrderingProperty` method should handle this case.

Reply:
I've fixed the issue by changing the input of `putOrderingProperty`, is this a viable solution?